### PR TITLE
Migrate memory_controller.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 1]
```diff
@@ -158,32 +159,29 @@ class AllocationStackTrace {
   String get sourcesDisplay => sources.join('\r');
 
   void computeStacktrace(CpuSample cpuSample,
-      {List<ProfileFunction> functions}) {
-    if (cpuSample.stack.isNotEmpty) {
-      final stackLength = cpuSample.stack.length;
+      {List<ProfileFunction>? functions}) {
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:162]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 2]
```diff
@@ -196,7 +194,8 @@ class AllocationStackTrace {
           } else {
             stacktrace.add('$ownerName.$functionName');
           }
-          sources.add(profileFunc.resolvedUrl);
+          if (profileFunc.resolvedUrl != null)
+            sources.add(profileFunc.resolvedUrl!);
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:198]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 3]
```diff
@@ -450,18 +451,18 @@ class MemoryController extends DisposableController
...
   static const liveFeed = 'Live Feed';
 
-  String memorySourcePrefix;
+  String? memorySourcePrefix;
 
   /// Notifies that the source of the memory feed has changed.
   ValueListenable get memorySourceNotifier => _memorySourceNotifier;
 
-  final _memorySourceNotifier = ValueNotifier<String>(liveFeed);
+  final _memorySourceNotifier = ValueNotifier<String?>(liveFeed);
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:459]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 4]
```diff
@@ -53,32 +54,32 @@ enum ChartInterval {
...
+  chartDuration(ChartInterval.FiveMinutes)!.inMinutes.toString(),
+  chartDuration(ChartInterval.TenMinutes)!.inMinutes.toString(),
   displayAll,
 ];
 
 String displayDuration(ChartInterval interval) =>
     displayDurationsStrings[interval.index];
 
-ChartInterval chartInterval(String displayName) {
+ChartInterval? chartInterval(String displayName) {
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:82]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 5]
```diff
@@ -496,29 +498,29 @@ class MemoryController extends DisposableController
   /// Default is to display default tick width based on width of chart of the collected
   /// data in the chart.
   final _displayIntervalNotifier =
-      ValueNotifier<ChartInterval>(ChartInterval.Default);
+      ValueNotifier<ChartInterval?>(ChartInterval.Default);
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:501]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 6]
```diff
@@ -621,12 +622,12 @@ class MemoryController extends DisposableController
 
   /// Track where/when a class is allocated (constructor new'd).
   Future<void> _setTracking(ClassRef ref, bool enable) async {
-    if (!await isIsolateLive(_isolateId)) return;
+    if (!await isIsolateLive(_isolateId!)) return;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:625]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 7]
```diff
@@ -642,7 +643,7 @@ class MemoryController extends DisposableController
         return;
       }
       // Add to tracking list.
-      trackAllocations[ref.name] = ref;
+      trackAllocations[ref.name!] = ref;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:646]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 8]
```diff
@@ -799,57 +795,57 @@ class MemoryController extends DisposableController
   static const groupByClass = 'Class';
   static const groupByInstance = 'Instance';
 
-  final groupingBy = ValueNotifier<String>(groupByLibrary);
+  final groupingBy = ValueNotifier<String?>(groupByLibrary);
 
-  ValueListenable<String> get groupingByNotifier => groupingBy;
+  ValueListenable<String?> get groupingByNotifier => groupingBy;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:800]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 9]
```diff
@@ -1000,11 +997,11 @@ class MemoryController extends DisposableController
   Future<List<ClassHeapDetailStats>> getAllocationProfile({
     bool reset = false,
   }) async {
-    if (!await isIsolateLive(_isolateId)) return [];
+    if (!await isIsolateLive(_isolateId!)) return [];
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:1000]: done

File: packages/devtools_app/lib/src/screens/memory/memory_controller.dart [hunk 10]
```diff
@@ -1133,7 +1131,7 @@ class MemoryController extends DisposableController
 
       // Sum up the externalSize of the children, under the externalReference group.
       externalReference.sumExternalSizes +=
-          liveExternal.externalProperty.externalSize;
+          liveExternal.externalProperty.externalSize as int;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_controller.dart:1134]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**